cmd/instances.go

```diff
@@ -0,0 +1,254 @@
+		if err != nil {
+			return err
+		}
+
+		if strings.ToLower(strings.TrimSpace(in)) != "yes" {
+			return nil
+		}
+
+		fmt.Println()
+
+		i, err := instancesRequest("DELETE", "/instances/"+url.QueryEscape(domain), nil, nil)
+		if err != nil {
+			log.Errorf("Error while removing instance for domain %s", domain)
+			return err
+		}
+
+		fmt.Println()
+
+		log.Infof("Instance for domain %s has been destroyed with success", i.Attrs.Domain)
+		return nil
+	},
+}
+
+var passwdInstanceCmd = &cobra.Command{
+	Use:   "passwd [directory]",
+	Short: "Generate an admin password",
+	Long: `
+cozy-stack instances password generate a password hash and
+print it to stdout.
+`,
+	RunE: func(cmd *cobra.Command, args []string) error {
+		if len(args) == 0 {
+			return cmd.Help()
+		}
+
+		directory := filepath.Join(config.AbsPath(args[0]))
+		info, err := os.Stat(directory)
+		if err != nil {
+			return err
+		}
+		if !info.IsDir() {
+			return fmt.Errorf("%s is not a directory", directory)
+		}
+
+		fmt.Fprintf(os.Stdout, "Password:")
+		password1, err := gopass.GetPasswd()
+		if err != nil {
+			return err
+		}
+
+		fmt.Fprintf(os.Stdout, "Confirmation:")
+		password2, err := gopass.GetPasswd()
+		if err != nil {
+			return err
+		}
+		if !bytes.Equal(password1, password2) {
+			return fmt.Errorf("Password missmatch")
+		}
+
+		b, err := crypto.GenerateFromPassphrase(password1)
+		if err != nil {
+			return err
+		}
+
+		filename := filepath.Join(directory, config.AdminSecretFileName)
+		f, err := os.OpenFile(filename, os.O_WRONLY|os.O_CREATE|os.O_TRUNC, 0444)
+		if err != nil {
+			return err
+		}
+		defer f.Close()
+
+		if err = os.Chmod(filename, 0444); err != nil {
+			return err
+		}
+
+		_, err = fmt.Fprintln(f, string(b))
+		if err != nil {
+			return err
+		}
+
+		fmt.Println("Hashed password outputed in", filename)
+		return nil
+	},
+}
+
+type instanceData struct {
+	ID    string             `json:"id"`
+	Rev   string             `json:"rev"`
+	Attrs *instance.Instance `json:"attributes"`
+}
+
+type instanceAPIData struct {
+	Data *instanceData `json:"data"`
+}
+
+type instancesAPIData struct {
+	Data []*instanceData `json:"data"`
+}
+
+func instancesClient() *client {
+	var password []byte
+
+	if !config.IsDevRelease() {
+		var err error
+		fmt.Printf("Password:")
+		password, err = gopass.GetPasswdMasked()
+		if err != nil {
+			panic(err)
+		}
+	}
+
+	return &client{
+		addr: config.AdminServerAddr(),
+
+		username: "",
+		password: string(password),
+	}
+}
+
+func instancesRequest(method, path string, q url.Values, body interface{}) (*instanceData, error) {
+	var doc instanceAPIData
+	err := clientRequestParsed(instancesClient(), method, path, q, body, &doc)
+	if err != nil {
+		return nil, err
+	}
+	return doc.Data, nil
+}
+
+func init() {
+	instanceCmdGroup.AddCommand(addInstanceCmd)
+	instanceCmdGroup.AddCommand(lsInstanceCmd)
+	instanceCmdGroup.AddCommand(destroyInstanceCmd)
+	instanceCmdGroup.AddCommand(passwdInstanceCmd)
```

Comment:
yes I agree its a bit weird under the instances command... you think it should be a root command ?